### PR TITLE
Allow for lists within the command chain

### PR DESCRIPTION
**Pull request by mstabrin**
This change allows to proper visualize cli information if the nesting contains lists.

### CODE REVIEW

File: npe2/cli.py [hunk 1]
```diff
@@ -231,7 +246,7 @@ def _make_rows(pm_dict: dict, normed_fields: Sequence[str]) -> Iterator[List]:
 @app.command()
 def list(
     fields: str = typer.Option(
-        "name,version,npe2,contributions",
+        "name,version,npe2,contributions,contributions.widgets.display_name",
```
Comment (tlambert03) [inline npe2/cli.py:249]: i don't think the default value should be changed here.  too specific to a particular use case.  If you want to get all the widget display names, you should use the `fields` option of `npe2 list` manually

Comment (mstabrin) [reply npe2/cli.py]: Yes indeed, I reverted the change :)

File: src/npe2/cli.py [hunk 2]
```diff
@@ -105,6 +105,9 @@ def _pprint_table(
             val = ""
             if isinstance(r, dict):
                 val = ", ".join(f"{k} ({v})" for k, v in r.items())
+            # list if overridden by function name
+            elif isinstance(r, type([])):
+                val = ", ".join(str(_) for _ in r)
```
Comment (Czaki) [inline src/npe2/cli.py:110]:
```suggestion
            elif isinstance(r, list):
                val = ", ".join(str(text) for text in r)
```

`_` means in python unused variable

File: src/npe2/cli.py [hunk 3]
```diff
@@ -212,9 +215,21 @@ def _make_rows(pm_dict: dict, normed_fields: Sequence[str]) -> Iterator[List]:
...
-                while parts and hasattr(val, "__getitem__"):
-                    val = val[parts.pop(0)]
+                val = [info]
+                while parts:
+                    part = parts.pop(0)
+                    new_val = []
+                    for cur_val in val:
+                        if hasattr(cur_val, "__getitem__"):
+                            # list if overridden by function name
+                            if isinstance(cur_val[part], type([])):
```
Comment (Czaki) [inline src/npe2/cli.py:225]:
```suggestion
                            if isinstance(cur_val[part], list):
```

File: src/npe2/cli.py [hunk 4]
```diff
@@ -212,9 +215,21 @@ def _make_rows(pm_dict: dict, normed_fields: Sequence[str]) -> Iterator[List]:
...
+                                new_val.extend(cur_val[part])
+                            else:
+                                new_val.append(cur_val[part])
+                    val = new_val
```
Comment (Czaki) [inline src/npe2/cli.py:229]: This does not look like reproduce and extend of the previous logic. It may break tests.

**Comment by jni** [comment]
@mstabrin do you think you might have a chance to pick this up again soon? 🙏

**Comment by nclack** [comment]
I'm going to go ahead and close this since there hasn't been activity for a while. Can always reopen if there's more interest/activity.

**Pull request closed without merging**